### ISSUE TITLE
Setup of KNIME SDK for KNIME AP 4.2 to work with "knip-omero"

### ISSUE THREAD

**Issue by charismatic-claire** [knime-ip/knip-omero#20]
Hi, I'm trying to setup a KNIME SDK in Eclipse 2020-09 for RCP and RAP developers so I can inspect and work with the nodes in this project (knip-omero). Unfortunately that does not work very well. Let me describe my setup real quick:

First of all, I'm using this target platform definition, inspired by the "knime-sdk-setup" and the "knip-sdk-setup" repositories:

```
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<?pde version="3.8"?>
<target name="KNIME Analytics Platform (4.2 release)" sequenceNumber="1512743963">
<locations>

<!-- default stuff-->
<location includeAllPlatforms="false" includeConfigurePhase="false" includeMode="planner" includeSource="true" type="InstallableUnit">
	<unit id="com.knime.features.explorer.serverspace.feature.group" version="0.0.0"/>
	<unit id="org.knime.features.base.feature.group" version="0.0.0"/>
	<unit id="org.knime.features.database.feature.group" version="0.0.0"/>
	<unit id="org.knime.features.dbdrivers.feature.group" version="0.0.0"/>
	<unit id="org.knime.features.ensembles.feature.group" version="0.0.0"/>
	<unit id="org.knime.features.ext.itemset.feature.group" version="0.0.0"/>
	<unit id="org.knime.features.ext.jep.feature.group" version="0.0.0"/>
	<unit id="org.knime.features.ext.poi.feature.group" version="0.0.0"/>
	<unit id="org.knime.features.js.quickforms.feature.group" version="0.0.0"/>
	<unit id="org.knime.features.json.feature.group" version="0.0.0"/>
	<unit id="org.knime.features.personalproductivity.feature.group" version="0.0.0"/>
	<unit id="org.knime.features.testing.application.feature.group" version="0.0.0"/>
	<unit id="org.knime.features.xml.feature.group" version="0.0.0"/>
	<unit id="org.knime.targetPlatform.feature.group" version="0.0.0"/>
	<unit id="org.knime.features.core.feature.group" version="0.0.0"/>
	<unit id="org.knime.features.javasnippet.feature.group" version="0.0.0"/>
	<unit id="org.knime.features.js.core.feature.group" version="0.0.0"/>
	<unit id="org.knime.features.product.feature.group" version="0.0.0"/>
	<unit id="org.knime.features.stats.feature.group" version="0.0.0"/>
	<unit id="org.knime.features.timeseries.feature.group" version="0.0.0"/>
	<unit id="org.knime.features.workbench.feature.group" version="0.0.0"/>
	<repository location="https://update.knime.com/analytics-platform/4.2/"/>
</location>

<!-- knip extensions -->
<location includeAllPlatforms="false" includeConfigurePhase="false" includeMode="planner" includeSource="true" type="InstallableUnit">
	<repository location="https://update.knime.com/community-contributions/trusted/4.2"/>
	<unit id="org.knime.knip.dl.feature.feature.group" version="1.0.2.v201907171233"/>
	<unit id="org.knime.knip.dl4j.feature.feature.group" version="1.1.3.v201907171244"/>
	<unit id="org.knime.knip.feature.feature.group" version="1.8.3.202009230803"/>
	<unit id="org.knime.knip.knimepython.feature.feature.group" version="1.0.9.v202007090959"/>
	<!-- <unit id="org.knime.knip.omero.feature.feature.group" version="0.11.0.v202006260943"/> -->
	<unit id="org.knime.knip.knimepython.source.feature.feature.group" version="1.0.9.v202007090959"/>
	<!-- <unit id="org.knime.knip.omero.source.feature.feature.group" version="0.11.0.v202006260943"/> -->
	<unit id="org.knime.knip.source.feature.feature.group" version="1.8.3.202009230803"/>
</location>

<!-- knip dependencies -->
<location includeAllPlatforms="false" includeConfigurePhase="false" includeMode="planner" includeSource="true" type="InstallableUnit">
<repository location="https://community.knime.org/download/knip-externals-trunk"/>
  <unit id="xml-apis" version="1.4.1.202007061359"/>
  <unit id="imagej-deprecated" version="0.1.2.202007061359"/>
  <unit id="jython-shaded" version="2.7.1.202007061359"/>
  <unit id="scijava-cache" version="0.1.2.202007061359"/>
  <unit id="scripting-javascript" version="0.4.4.202007061359"/>
  <unit id="args4j" version="2.0.29.202007061359"/>
  <unit id="ij" version="1.52.0.202007061359"/>
  <unit id="ilastik4ij" version="1.0.3.202007061359"/>
  <unit id="imagej-common" version="0.28.1.202007061359"/>
  <unit id="imagej-ops" version="0.44.0.202007061359"/>
  <unit id="imagej-plugins-commands" version="0.8.1.202007061359"/>
  <unit id="imagej-ui-awt" version="0.3.1.202007061359"/>
  <unit id="imagej-ui-swing" version="0.22.0.202007061359"/>
  <unit id="imglib2" version="5.8.0.202007061359"/>
  <unit id="imglib2-algorithm" version="0.11.1.202007061359"/>
  <unit id="imglib2-algorithm-fft" version="0.2.0.202007061359"/>
  <unit id="imglib2-algorithm-gpl" version="0.2.3.202007061359"/>
  <unit id="imglib2-ij" version="2.0.44.202007061359"/>
  <unit id="imglib2-realtransform" version="2.1.0.202007061359"/>
  <unit id="imglib2-roi" version="0.7.0.202007061359"/>
  <unit id="imglib2-ui" version="2.0.0.202007061359"/>
  <unit id="jep" version="2.4.2.202007061359"/>
  <unit id="knip-tmp-imglib2-ops" version="0.4.1.202007061359"/>
  <unit id="knip-trackmate-fork" version="2.7.4.202007061359"/>
  <unit id="miglayout" version="3.7.4.202007061359"/>
  <unit id="scifio" version="0.37.3.202007061359"/>
  <unit id="scifio-bf-compat" version="3.0.0.202007061359"/>
  <unit id="scifio-lifesci" version="0.8.0.202007061359"/>
  <unit id="scijava-common" version="2.77.0.202007061359"/>
  <unit id="scijava-plugins-commands" version="0.2.3.202007061359"/>
  <unit id="scijava-plugins-text-markdown" version="0.1.3.202007061359"/>
  <unit id="scijava-plugins-text-plain" version="0.1.3.202007061359"/>
  <unit id="scijava-ui-awt" version="0.1.6.202007061359"/>
  <unit id="scijava-ui-swing" version="0.12.0.202007061359"/>
  <unit id="scripting-java" version="0.4.1.202007061359"/>
  <unit id="scripting-jython" version="0.4.2.202007061359"/>
  <unit id="scripting-beanshell" version="0.3.3.202007061359"/>
  <unit id="scripting-clojure" version="0.1.6.202007061359"/>
  <unit id="scripting-groovy" version="0.2.7.202007061359"/>
  <unit id="scripting-jruby" version="0.3.0.202007061359"/>
  <unit id="bigdataviewer-core" version="5.1.0.202007061359"/>
  <unit id="bigdataviewer-vistools" version="1.0.11.202007061359"/>
  <unit id="ui-behaviour" version="1.7.4.202007061359"/>
  <unit id="spim_data" version="2.2.2.202007061359"/>
  <unit id="jfreechart" version="1.5.0.202007061359"/>
  <unit id="imagej-updater" version="0.9.3.202007061359"/>
  <unit id="script-editor" version="0.4.0.202007061359"/>
  <unit id="parsington" version="1.0.4.202007061359"/>
  <unit id="eventbus" version="1.4.0.202007061359"/>
  <unit id="imagej-mesh" version="0.7.1.202007061359"/>
  <unit id="scijava-search" version="0.6.0.202007061359"/>
  <unit id="trove4j" version="3.0.3.202007061359"/>
  <unit id="imglib2-cache" version="1.0.11.202007061359"/>
  <unit id="eventbus.source" version="1.4.0.202007061359"/>
  <unit id="imagej-mesh.source" version="0.7.1.202007061359"/>
  <unit id="imglib2-cache.source" version="1.0.11.202007061359"/>
  <unit id="scijava-search.source" version="0.6.0.202007061359"/>
  <unit id="trove4j.source" version="3.0.3.202007061359"/>
  <unit id="parsington.source" version="1.0.4.202007061359"/>
  <unit id="imagej-updater.source" version="0.9.3.202007061359"/>
  <unit id="jfreechart.source" version="1.5.0.202007061359"/>
  <unit id="script-editor.source" version="0.4.0.202007061359"/>
  <unit id="args4j.source" version="2.0.29.202007061359"/>
  <unit id="bigdataviewer-core.source" version="5.1.0.202007061359"/>
  <unit id="ilastik4ij.source" version="1.0.3.202007061359"/>
  <unit id="imagej-common.source" version="0.28.1.202007061359"/>
  <unit id="imagej-deprecated.source" version="0.1.2.202007061359"/>
  <unit id="imagej-ops.source" version="0.44.0.202007061359"/>
  <unit id="imagej-plugins-commands.source" version="0.8.1.202007061359"/>
  <unit id="imagej-ui-awt.source" version="0.3.1.202007061359"/>
  <unit id="imglib2-algorithm-fft.source" version="0.2.0.202007061359"/>
  <unit id="imglib2-algorithm.source" version="0.11.1.202007061359"/>
  <unit id="imglib2-realtransform.source" version="2.1.0.202007061359"/>
  <unit id="imglib2-roi.source" version="0.7.0.202007061359"/>
  <unit id="imglib2-ui.source" version="2.0.0.202007061359"/>
  <unit id="imglib2.source" version="5.8.0.202007061359"/>
  <unit id="jep.source" version="2.4.2.202007061359"/>
  <unit id="knip-tmp-imglib2-ops.source" version="0.4.1.202007061359"/>
  <unit id="miglayout.source" version="3.7.4.202007061359"/>
  <unit id="scifio-lifesci.source" version="0.8.0.202007061359"/>
  <unit id="scifio.source" version="0.37.3.202007061359"/>
  <unit id="scijava-cache.source" version="0.1.2.202007061359"/>
  <unit id="scijava-common.source" version="2.77.0.202007061359"/>
  <unit id="scijava-plugins-commands.source" version="0.2.3.202007061359"/>
  <unit id="scijava-plugins-text-markdown.source" version="0.1.3.202007061359"/>
  <unit id="scijava-plugins-text-plain.source" version="0.1.3.202007061359"/>
  <unit id="scijava-ui-awt.source" version="0.1.6.202007061359"/>
  <unit id="scijava-ui-swing.source" version="0.12.0.202007061359"/>
  <unit id="scripting-beanshell.source" version="0.3.3.202007061359"/>
  <unit id="scripting-clojure.source" version="0.1.6.202007061359"/>
  <unit id="scripting-groovy.source" version="0.2.7.202007061359"/>
  <unit id="scripting-java.source" version="0.4.1.202007061359"/>
  <unit id="scripting-javascript.source" version="0.4.4.202007061359"/>
  <unit id="spim_data.source" version="2.2.2.202007061359"/>
  <unit id="ui-behaviour.source" version="1.7.4.202007061359"/>
  <unit id="xml-apis.source" version="1.4.1.202007061359"/>
  <unit id="scijava-table" version="0.4.0.202007061359"/>
  <unit id="scijava-table.source" version="0.4.0.202007061359"/>
</location>
</locations>
<targetJRE path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
<launcherArgs>
<vmArgs>-server
-Dsun.java2d.d3d=false
-Dosgi.classloader.lock=classname
-XX:+UnlockDiagnosticVMOptions
-XX:+UnsyncloadClass 
-XX:+UseG1GC
-ea
-Dorg.eclipse.swt.internal.gtk.cairoGraphics=false
-Dorg.eclipse.swt.internal.gtk.useCairo=false
-Dorg.eclipse.swt.browser.IEVersion=11001
-Dsun.awt.noerasebackground=true
-Dsun.net.client.defaultReadTimeout=0
-Dorg.eclipse.swt.internal.gtk.disablePrinting</vmArgs>
</launcherArgs>
</target>
```

I also installed all kinds projects I guessed to be dependencies needed. Have a look at this:

![setup](https://user-images.githubusercontent.com/44076467/96772790-a434e080-13b1-11eb-9c6e-a4115019dc16.png)

That almost works, but I keep getting this error message regarding line 78 in "org.knime.knip.omero2.nodes.tablereader.OmeroTableReaderNodeModel":

```
The type net.imagej.table.Table cannot be resolved. It is indirectly referenced from required .class files
```

The line of code in question is this:

```
table = m_omeroService.downloadTable(creds, m_tableIdModel.getIntValue());
```
So the ''downloadTable()" method of the "OMEROService" instance is not happy. I do not really know why. The whole thing looks like this: 

![error](https://user-images.githubusercontent.com/44076467/96771001-1eb03100-13af-11eb-8cf9-8ac25c16f5ae.png)

When I comment out the line in question and make the error dissapear, I get another error when I try to instanciate the node in KNIME AP. It looks like this:

![error2](https://user-images.githubusercontent.com/44076467/96771406-99794c00-13af-11eb-9d2c-6d347ecfc17b.png)

I'm not really sure why that is. After all, at least I can see all the Nodes, which is not the case when I just download them from the trusted community repository in KNIME AP 4.2.2. See how it looks like in KNIME AP 4.2.2

![repo1](https://user-images.githubusercontent.com/44076467/96771902-55d31200-13b0-11eb-8816-394764e1027b.png)

And this is what I see when I run the target platform as mentioned above:

![repo2](https://user-images.githubusercontent.com/44076467/96771967-6b483c00-13b0-11eb-87da-13925d24a5c4.png)

To summarize my question: Can you give a detailed description on how to setup the KNIME SDK correctly so that the current version of this project works for a target paltform in Eclipse for the current version of KNIME AP 4.2.2? That would be most helpful!
